core/src/main/java/org/elasticsearch/gateway/TransportNodesListGatewayStartedShards.java

```diff
@@ -123,7 +126,7 @@ protected NodeGatewayStartedShards nodeOperation(NodeRequest request) {
                     // we may send this requests while processing the cluster state that recovered the index
                     // sometimes the request comes in before the local node processed that cluster state
                     // in such cases we can load it from disk
-                    metaData = IndexMetaData.FORMAT.loadLatestState(logger, nodeEnv.indexPaths(shardId.getIndex()));
+                    metaData = metaStateService.getIndexMetadatFormat().loadLatestState(logger, nodeEnv.indexPaths(shardId.getIndex()));
```

Comment:
maybe just make this a public method on metaStateService? (i.e., loadIndexLatestState) it already has nodeEnv and everything it needs. And now that I check it - it already has one :D org.elasticsearch.gateway.MetaStateService#loadIndexState